angular-primeng-app/src/app/app.component.ts

```diff
@@ -50,9 +54,30 @@ export class AppComponent implements OnInit, OnDestroy {
 					});
 				}
 			});
+
+		this.scrollEvntSub = fromEvent(this.document, 'scroll')
+			.subscribe(() => {
+				if (this.calculatePositionScroll() > 50)
+					this.enableScrollUp = true;
+				else
+					this.enableScrollUp = false;
+			})
+	}
+
```

Comment:
@DongreJaipal and over here the using the logic you suggested, which looks good, we can use the `HostListener` which needs to be added as an import from '@angular/core' and can be used like this:

```
onWindowScroll() {
    // Check the vertical scroll position
    const yOffset = window.pageYOffset || document.documentElement.scrollTop;
    if (yOffset > 300) { // this value might need adjustments
      this.showScrollButton = true;
    } else {
      this.showScrollButton = false;
    }
  }
```
and then bellow it, the click event function 

```
  scrollToTop() {
    this.viewportScroller.scrollToPosition([0, 0]);
  }
```
please give it a try and let me know if you need any further assistance